socketio/src/client/builder.rs

```diff
@@ -30,15 +31,20 @@ pub enum TransportType {
 /// configuring the callback, the namespace and metadata of the socket. If no
 /// namespace is specified, the default namespace `/` is taken. The `connect` method
 /// acts the `build` method and returns a connected [`Client`].
+#[derive(Clone)]
 pub struct ClientBuilder {
     address: String,
-    on: HashMap<Event, Callback<SocketCallback>>,
-    on_any: Option<Callback<SocketAnyCallback>>,
+    on: Arc<RwLock<HashMap<Event, Callback<SocketCallback>>>>,
+    on_any: Arc<RwLock<Option<Callback<SocketAnyCallback>>>>,
     namespace: String,
     tls_config: Option<TlsConnector>,
     opening_headers: Option<HeaderMap>,
     transport_type: TransportType,
     auth: Option<serde_json::Value>,
+    pub reconnect: bool,
+    pub reconnect_attempts: u8, // 0 means infinity
```

Comment:
This is the maximum amount of allowed reconnects, the variable name suggests it is a counter, wdyt about renaming the variable to something like `max_reconnect_attempts`?

Reply:
I borrowed from the js client, but I think max_reconnect_attempts is better